### DISCUSSION TITLE
Can't get the APIView HTML form to show up

### DISCUSSION THREAD

**Discussion by RonyKordahi** [encode/django-rest-framework#9036] [Potential Issue] [answered]
I recently got a new job and I'm learning Django for it.

I've been following this [tutorial](https://www.youtube.com/watch?v=n2T9rmFmo48) on how to connect it to a React frontend. There is a shorter and more recent version of this tutorial [here](https://www.youtube.com/watch?v=fBA-jaWab9k).

In [this](https://www.youtube.com/watch?v=fBA-jaWab9k&t=1028s) part of the first linked video above, he sets up a class based view using `APIView` imported from `rest_framework.views`. However, I can't seem to get the HTML form to generate on my view page. My code is nearly identical to his (with a few changes to the name of the model and some variables) and I've tried everything I could find online but nothing's worked.

Not sure what I'm missing.

My app's models.py:
```
from django.db import models

# Create your models here.

class Employees(models.Model):
    employee = models.CharField(max_length=30)
    department = models.CharField(max_length=200)
```

My app's serializers.py:
```
from rest_framework import serializers

from .models import *

class EmployeesSerializer(serializers.ModelSerializer):

    class Meta:
        model = Employees
        fields = ["employee", "department"]
```

My app's views.py:
```
from django.shortcuts import render
from rest_framework.views import APIView
from rest_framework.response import Response

from .models import *
from .serializers import *

# Create your views here.

class EmployeesView(APIView):

    # returns all employees
    def get(self, req):
        result = [
            {"employee": output.employee, "department": output.department}
            for output in Employees.objects.all()
        ]

        return Response(result)
    
    # saves new employee
    def post(self, req):
        serializer = EmployeesSerializer(data=req.data)
        
        if serializer.is_valid(raise_exception=True):
            serializer.save()
            return Response(serializer.data)
```

My project's urls.py:
```
from django.contrib import admin
from django.urls import path, include
from api.views import *

urlpatterns = [
    path('admin/', admin.site.urls),
    path("", EmployeesView.as_view(), name="all_employees")
]
```

What gets generated by the view: 
![image](https://github.com/encode/django-rest-framework/assets/60363887/7fb0c2ba-7edd-47c5-80fe-d0bb7c054c44)

EDIT: modified code blocks so they would show up correctly
EDIT 2: added urls.py

**Comment by b4thestorm** [comment, marked as answer]
If you are building an API, then this should be expected. You have likely navigated to the DRF endpoint which is giving you a nice view of the contents of your employee model, serializer and Class View code for the GET endpoint specifically. What you would want to do is set up your react side of things to consume the contents of the APIView you just made. Use a tool like Axios or Fetch and hit the endpoint you have here and it will return to you the employees you have in your Employee model. Which it also. looks like you may want to seed your DB with some employees.

**Reply by RonyKordahi** [reply]
I haven't actually connected React to it yet, in the tutorial they just went to the view and the HTML form was there, but on my end I couldn't get it to appear at all. From my understanding, the APIView class generates the HTML form automatically, which lets you create a new instance of the model when you fill it up and submit it.

**Comment by DiabeDoumbia** [comment]
I know it bit late to respond at this problem but for everyone in future who came across a similar issue when using DRF with ApiView, if the form doesn't show up you will need to check your ApiView class and make sure to have : 

class YourApiVIewClass(APIView):

    serializer_class = YourApiSerializer      

    def post(self,request):
         (your post request code goes here)

    def get(self,request):
         (your get request code goes here)
